src/components/App.jsx

```diff
@@ -1,43 +1,81 @@
 /* global cozy */
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import { PropTypes } from 'prop-types'
 import flow from 'lodash/flow'
 
+import { useClient } from 'cozy-client'
 import { Main, Layout } from 'cozy-ui/transpiled/react/Layout'
 import Alerter from 'cozy-ui/transpiled/react/Alerter'
 import { Title } from 'cozy-ui/transpiled/react/Text'
 import { useI18n, useBreakpoints } from 'cozy-ui/transpiled/react'
 import { Sprite as IconSprite } from 'cozy-ui/transpiled/react/Icon'
 import flag, { FlagSwitcher } from 'cozy-flags'
 import 'cozy-ui/transpiled/react/stylesheet.css'
+import log from 'cozy-logger'
 
 import withContactsMutations from '../connections/allContacts'
 import ContactsSelectionBar from './layout/ContactsSelectionBar'
 import { ModalManager } from '../helpers/modalManager'
-import { initFlags } from '../helpers/flags'
+import useFlags from './Hooks/useFlags'
 import container from './AppContainer'
 import ContentWrapper from './ContentWrapper'
+import SpinnerContact from './Components/Spinner'
+import { fetchNormalizedServiceByName } from '../helpers/fetches'
 
 const ContactsApp = props => {
   // HACK to avoid CozyBar error :
   // you tried to use the CozyBar API (BarCenter) but the CozyBar is not initialised yet via cozy.bar.init
   // TODO : TO BE REMOVED
   const [cozyBarHack, setcozyBarHack] = useState(false)
 
+  const [serviceToLaunch, setServiceToLaunch] = useState(null)
+  const [hasServiceBeenLaunched, setHasServiceBeenLaunched] = useState(null)
+
+  useFlags()
+  const client = useClient()
+  const { t } = useI18n()
+  const { isMobile } = useBreakpoints()
+  const { BarCenter } = cozy.bar
+  const { deleteContact, cleanTrashedGroups } = props
+
+  const setStateOfServiceToLaunch = useCallback(async () => {
+    const serviceToLaunch = await fetchNormalizedServiceByName(
+      client,
+      'keepIndexFullNameAndDisplayNameUpToDate'
+    )
+    setServiceToLaunch(serviceToLaunch)
+    setHasServiceBeenLaunched(
+      serviceToLaunch.current_state.last_success.length > 0
+    )
+  }, [])
+
   useEffect(() => {
-    initFlags()
-    props.cleanTrashedGroups()
+    cleanTrashedGroups()
+  }, [])
 
+  useEffect(
+    () => {
+      setStateOfServiceToLaunch()
+    },
+    [setStateOfServiceToLaunch]
+  )
```

Comment:
Pourquoi as-tu utilisé `useCallback` sur cet effet?

Reply:
C'est pour éviter que la fonction appelée par l'effet soit récrée à chaque nouveau rendu du composant

Reply:
Hmm, le useEffect est exécuté que quand ses dépendances changent, non ? https://fr.reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
Oui la fonction n'est pas ré-exécutée mais re-créée (https://itnext.io/write-clean-er-components-jsx-1e70491baded point 5)

Reply:
OK, so I that case, we should add `client` to the list of the dependencies array of the useCallback. Because the client can change

Reply:
Ah yes good catch thanks